src/Grpc.AspNetCore.Server/Internal/GrpcServiceOptionsSetup.cs

```diff
@@ -33,6 +35,10 @@ public void Configure(GrpcServiceOptions options)
             {
                 options.ReceiveMaxMessageSize = DefaultReceiveMaxMessageSize;
             }
+            if (options._compressionProviders == null || options._compressionProviders.Count == 0)
```

Comment:
if we are going to add a default, is there any reason to optimize for the case where _compressionProvider is null? I think _compressionProviders should be set to private and just initialize it as an empty list.

Reply:
A null value is needed for inheritance from the global settings to the per-service settings.